functional/security/src/net/adoptopenjdk/test/Tls13FunctionalTest.java

```diff
@@ -0,0 +1,228 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.adoptopenjdk.test;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import org.testng.log4testng.Logger;
+import javax.net.ssl.HandshakeCompletedEvent;
+import javax.net.ssl.HandshakeCompletedListener;
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLPeerUnverifiedException;
+import javax.net.ssl.SSLSocket;
+import javax.net.ssl.SSLSocketFactory;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.Socket;
+import java.net.SocketTimeoutException;
+import java.net.URL;
+import java.net.UnknownHostException;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Creates actual HTTPS connections to real webservers to exercise the TLS stack. Connections that do not use TLSv1.3
+ * are rejected.
+ * <p>
+ * TLSv1.3 was introduced with OpenJDK 11 (see https://openjdk.java.net/jeps/332) and backported to OpenJDK 8u272
+ * (https://bugs.openjdk.java.net/browse/JDK-8245466). Tests are automatically skipped on versions that do not support
+ * TLSv1.3.
+ */
```

Comment:
Do we want to track authorship as part of the file? In that case, we'd need:

```suggestion
 *
 * @author Andreas Ahlenstorf, Shelley Lambert
 */
```

Reply:
We have not been tracking authorship in this way, but I should figure out how to put you as author or co-author on the git commit.